python/cuml/preprocessing/TargetEncoder.py

```diff
@@ -407,3 +407,19 @@ def _get_output_type(self, x):
                 or isinstance(x, pandas.Series):
             return 'numpy'
         return 'cupy'
+
+    def get_param_names(self):
+        return [
+            "n_folds", "smooth", "seed", "split_method",
+        ]
+
+    def get_params(self):
+        """
+        Returns a dict of all params owned by this class.
+        """
+        params = dict()
+        variables = self.get_param_names()
+        for key in variables:
+            var_value = getattr(self, key, None)
+            params[key] = var_value
+        return params
```

Comment:
Should be inheritable from the `cuml.common.base.Base` class.

Reply:
Thank you for the review. Yes, I tried inheriting `base` in the first place but I ended up making more changes than necessary to pass `test_base.py`.  I'll try again in another PR.